metals/src/main/scala/scala/meta/internal/metals/codeactions/FlatMapToForComprehensionCodeAction.scala

```diff
@@ -0,0 +1,503 @@
+      else None
+    )
+  }
+
+  private def processValueNameAndNextQual(
+      tree: Tree,
+      nameGenerator: MetalsNames
+  ): Option[(String, Term)] = {
+    tree match {
+      case Term.Function(List(param), term) if param.name.value.isEmpty =>
+        val newName =
+          nameGenerator.createNewName()
+        Some((newName, term))
+      case Term.Function(List(param), term) =>
+        (Some((param.name.value, term)))
+
+      case Term.AnonymousFunction(term) =>
+        replacePlaceHolderInTermWithNewName(term, nameGenerator)
+      case term: Term
+          if term.is[Term.Apply] || term.is[Term.Eta] || term
+            .is[Term.ApplyUnary] || term.is[Term.ApplyUsing] || term
+            .is[Term.ApplyInfix] || term.is[Term.Name] || term
+            .is[Term.Select] =>
+        replacePlaceHolderInTermWithNewName(term, nameGenerator)
+      case Term.Block(List(function)) =>
+        processValueNameAndNextQual(function, nameGenerator)
+      case _ =>
+        None
+
+    }
+  }
+
+  private def obtainNextYieldAndElemsForMap(
+      nameGenerator: MetalsNames,
+      perhapsLastName: Option[String],
+      shouldFlat: Boolean,
+      existingForElements: List[Enumerator],
+      currentYieldTerm: Option[Term],
+      nextQual: Term
+  ): (List[Enumerator], Option[Term]) = {
+    perhapsLastName match {
+      case Some(lastName) =>
+        (
+          List(
+            if (shouldFlat)
+              Enumerator.Generator.apply(
+                Pat.Var.apply(Term.Name.apply(lastName)),
+                nextQual
+              )
+            else
+              Enumerator.Val.apply(
+                Pat.Var.apply(Term.Name.apply(lastName)),
+                nextQual
+              )
+          ) ++ existingForElements,
+          currentYieldTerm
+        )
+      case None =>
+        if (shouldFlat) {
+          val lastGeneratedName =
+            nameGenerator.createNewName()
+          val newEnumerations = List(
+            Enumerator.Generator(
+              Pat.Var.apply(Term.Name.apply(lastGeneratedName)),
+              nextQual
+            )
+          )
+          val newYield = Term.Name.apply(lastGeneratedName)
+          (newEnumerations, Some(newYield))
+        } else (existingForElements, Some(nextQual))
+    }
+  }
+
+  private def processMap(
+      elems: List[Enumerator],
+      maybeYieldTerm: Option[Term],
+      nameGenerator: MetalsNames,
+      valueName: String,
+      termSelectQual: Term
+  ): (List[Enumerator], Option[Term]) = {
+
+    termSelectQual match {
+      case qualTermApply: Term.Apply =>
+        extractChainedForYield(
+          Some(valueName),
+          maybeYieldTerm,
+          elems,
+          qualTermApply,
+          nameGenerator
+        )
+      case otherQual =>
+        (
+          Enumerator.Generator(
+            Pat.Var.apply(Term.Name.apply(valueName)),
+            otherQual
+          )
+            +: elems,
```

Comment:
`+:` is a generic method for prepend, so it performs some additional calls.
It's a valid choice in case of having `Seq` but when you have a `List` it's more common just to use `::` constructor